server/routes/contacts/manage/contact-confirmation/contactConfirmationController.ts

```diff
@@ -4,19 +4,33 @@ import { Page } from '../../../../services/auditService'
 import { IsContactConfirmedSchema } from './contactConfirmationSchema'
 import { navigationForAddContactJourney, nextPageForAddContactJourney } from '../../add/addContactFlowControl'
 import PrisonerJourneyParams = journeys.PrisonerJourneyParams
+import Contact = contactsApiClientTypes.Contact
+import { ContactsService } from '../../../../services'
+import logger from '../../../../../logger'
 
 export default class ContactConfirmationController implements PageHandler {
+  constructor(private readonly contactsService: ContactsService) {}
+
   public PAGE_NAME = Page.CONTACT_CONFIRMATION_PAGE
 
   GET = async (
     req: Request<PrisonerJourneyParams, unknown, unknown, { contactId?: string }>,
     res: Response,
   ): Promise<void> => {
     const { journeyId } = req.params
-    const { prisonerDetails } = res.locals
+    const { prisonerDetails, user } = res.locals
     const journey = req.session.addContactJourneys[journeyId]
+    const validationErrors = res.locals.validationErrors?.isContactConfirmed
+    let contact: Contact
+
+    try {
+      contact = validationErrors ? null : await this.contactsService.getContact(journey.contactId, user)
+    } catch (error) {
+      logger.error(JSON.stringify(error))
+    }
```

Comment:
You don't need to catch errors here as the controller is wrapped with `asyncMiddleware` it should handle it correctly if the Promise is rejected.

I am also a bit confused about the logic. Doesn't this mean that if you submit the page without selecting the contact you will get validation errors but not populate the contact in the view model?

Reply:
Yes, you're right without selecting the contact the service was not executed and the page displayed without info (see video as well). Try/catch removed.